apps/schema2/src/components/SchemaBeta.vue

```diff
@@ -0,0 +1,240 @@
+    Spinner,
+    JsonView,
+  },
+  data() {
+    return {
+      schema: {},
+      loading: false,
+      graphqlError: null,
+      warning: null,
+      success: null,
+      showDiagram: false,
+    };
+  },
+  methods: {
+    saveSchema() {
+      this.graphqlError = null;
+      this.warning = "submitting changes";
+      let tables = this.schema.tables;
+      this.schema.tables = null; //hide it all
+
+      //transform subclasses back into their original tables.
+      //create a map of tables
+      let tableMap = tables.reduce((map, table) => {
+        map[table.name] = table;
+        table.subclasses.forEach((subclass) => (map[subclass.name] = subclass));
+        delete table.subclasses;
+        return map;
+      }, {});
+      //redistribute the columns to subclasses
+      tables.forEach((table) => {
+        table.columns.forEach((column) => {
+          if (column.table != table.name) {
+            tableMap[column.table].columns.push(column);
+          }
+        });
+      });
+      tables.forEach((table) => {
+        table.columns = table.columns.filter(
+          (column) => column.table === table.name
+        );
+      });
+      request(
+        "graphql",
+        `mutation change($tables:[MolgenisTableInput]){change(tables:$tables){message}}`,
+        {
+          tables: tables,
+        }
+      )
+        .then(() => {
+          this.warning = "submission complete, reloading schema";
+          this.loadSchema();
+          this.warning = null;
+          this.success = `Schema saved`;
+          this.warning = null;
+        })
+        .catch((error) => {
+          if (error.response.status === 403) {
+            this.graphqlError = "Forbidden. Do you need to login?";
+            this.showLogin = true;
+          } else {
+            this.graphqlError = error.response.errors[0].message;
+          }
+        });
+      this.loading = false;
+    },
+    loadSchema() {
+      this.graphqlError = null;
+      this.loading = true;
+      request(
+        "graphql",
+        "{_schema{name,tables{name,tableType,inherit,externalSchema,description,semantics,columns{name,table,position,columnType,inherited,key,refSchema,refTable,refLink,refBack,required,description,semantics,validation,visible}}}}"
+      )
+        .then((data) => {
+          let _schema = this.addOldNamesAndRemoveMeta(data._schema);
+          this.schema = this.convertToSubclassTables(_schema);
+        })
+        .catch((error) => {
+          this.graphqlError = error;
+          if (
+            this.graphqlError.includes(
+              "Field '_schema' in type 'Query' is undefined"
+            )
+          ) {
+            this.error =
+              "Schema is unknown or permission denied (might you need to login with authorized user?)";
+          }
+        })
+        .finally(() => {
+          this.loading = false;
+          this.warning = null;
+        });
+    },
+    addOldNamesAndRemoveMeta(schema) {
```

Comment:
needs unit test

Reply:
do we have standard way of working? moreover, this is so at the heart of the system that I really need to think hard what edge cases need testing.